### PR TITLE
feat(docs): automation pipeline — analyzers, generators, workflows, Phase 0 pilot

### PR DESCRIPTION
**Pull request by joshsmithxrm**
## Summary

Full implementation of the docs-automation pipeline per `specs/docs-generation.md`.

This PR delivers the spec's first 17 ACs (analyzer rules, generators, smoke, workflows, MDX+terminology validation) plus the Phase 0 curation pattern validated on `PPDS.Auth`. The remaining three library annotations (PPDS.Dataverse / PPDS.Migration / PPDS.Plugins) follow the same proven pattern but are scoped out to a follow-up PR (tracked as an issue).

### Workflow infrastructure fixes (2 bugs)
- `fix(workflow)` — `pr_monitor.py` used stale `gh pr checks --json conclusion` (replaced with `bucket`/`state` mapping); removed misleading `idle_prompt` Notification hook that fired "PR Ready" toasts for drafts
- `fix(start)` — `/start` skill rewritten to deliver handoff context inline via `pwsh Start-Process` + here-string (no more `.plans/context.md` files)

### Spec + design artifacts
- `spec(docs-generation)` — 549-line spec with 30 ACs; two bias-isolated review passes (spec + plan); design decisions cover Option D curation, PublicApiAnalyzers choice, custom Roslyn walker over DocFxMarkdownGen, CI-only smoke test, GitHub App over PAT, terminology (`libraries` not `SDK`)

### Phase 1 — Analyzers (PPDS014/015/016)
- **PPDS014 XmlDocOnPublicApi** — public types/members in the four libraries require `/// <summary>` or `[EditorBrowsable(Never)]`; scoped via assembly-name; skips `[GeneratedCode]` and `/Generated/`, `/obj/` paths — 6 tests
- **PPDS015 CliCommandNeedsDescription** — Spectre command subclasses require Description via attribute, property initializer, or ctor assignment — 4 tests
- **PPDS016 McpToolNeedsMetadata** — `[McpServerTool]` methods require `Name` and a description (inline named arg OR sibling `[System.ComponentModel.Description]` per codebase convention; falls back to syntax-tree inspection for partial compilations) — 3 tests

### Phase 2 — Four reference generators
- `scripts/docs-gen/PPDS.DocsGen.Common/` — shared infra: `IReferenceGenerator`, `MdxEscape.Prose` (fence- and inline-code-aware), `MdxEscape.InlineCode` (deterministic 3-tier wrapper satisfying AC-19), `BannerHelper` — 8 tests
- `cli-reflect/` — Spectre command tree → markdown via `MetadataLoadContext` — 3 tests
- `libs-reflect/` — Roslyn walker over library public surface with XML-doc parsing, filters `[EditorBrowsable(Never)]` silently, skips undocumented types with diagnostics — 4 tests
- `mcp-reflect/` + new `McpToolExampleAttribute` in `PPDS.Mcp` — enumerates `[McpServerTool]` methods, handles sibling `[Description]` pattern — 4 tests
- `ext-reflect/` (Node) — reads `package.json` `contributes` → three markdown files (commands, configuration, views) — 5 tests

All four generators are deterministic (byte-identical output on repeat runs verified per-generator).

### Phase 3 — Extension contribution lint
- `scripts/docs-gen/lint-extension-contributions.js` — Node CommonJS script checking every `contributes.commands` entry has `command`+`title`+`category`, wired into `scripts/hooks/pre-commit` — 2 tests

### Phase 4 — Smoke test tool
- `scripts/docs-gen/smoke/` — extracts fenced C# blocks from ppds-docs markdown, wraps in one of three forms (complete file / top-level statements / method body), compiles via Roslyn against current assemblies. Honors `// ignore-smoke` marker — 4 tests

### Phase 5 — Workflows + GitHub App helper
- `.github/workflows/docs-smoke.yml` — reusable `workflow_call` entry point from ppds-docs PRs
- `.github/workflows/docs-release.yml` — tag + `workflow_dispatch` dry-run support; wires all four generators + rollover scripts + app-token helper
- `scripts/docs-gen/app-token/` — C# JWT+installation-token helper (Constitution I1: stdout=token, stderr=diagnostics)
- `scripts/docs-gen/compute-rollover-diff.sh`, `check-open-rollover.sh`, `compute-surface-summary.sh` — building blocks tested directly (no `act` dependency) — 4 tests

### Cross-generator validation
- `MdxParseTests.StrictMdxAcceptsGeneratedReference` (AC-23) — parses every generator fixture through `@mdx-js/mdx` plus edge-case fixture covering `Task<T>`, `List<int>`, `Dictionary<string, List<Account>>`
- `TerminologyTests.NoSdkInProse` (AC-24) — scans for bare "SDK" outside fences, inline code, and Microsoft-adjacent contexts

### Phase 0 pilot — PPDS.Auth annotated + baseline
- 52 customer-facing types baselined (all already had `/// <summary>`), 2 marked `[EditorBrowsable(Never)]`, 1 `public` → `internal`
- `PPDS.Auth.csproj` adds `Microsoft.CodeAnalysis.PublicApiAnalyzers` + `<WarningsAsErrors>PPDS014;RS0016;RS0017</WarningsAsErrors>`
- `PublicAPI.Shipped.txt` seeded with 624 entries across three TFMs; `Unshipped.txt` empty
- Builds clean with WarningsAsErrors active across net8.0/net9.0/net10.0
- PPDS.Cli (2818 tests) + PPDS.Auth.Tests (421 tests) continue to pass

### Phase 6 — Integration polish
- `CLAUDE.md` — three new ALWAYS entries for PPDS014/015/016 so future sessions comply proactively
- `scripts/docs-gen/README.md` — overview, style guide, GitHub App setup runbook, baseline recovery, troubleshooting per analyzer
- `docs/runbooks/docs-release.md` — step-by-step release procedure, dry-run mode, failure recovery paths

## Testing

All new and existing tests pass across net8.0/net9.0/net10.0:
- Analyzer tests: 104 (13 existing + 13 new × 3 TFMs)
- Generator tests: 16 + 8 shared infra = 24
- Smoke tests: 4
- Workflow tests: 6 (2 cross-gen + 4 release/smoke orchestration)
- Extension lint tests: 2 (vitest)
- Full PPDS test suite: still green — no regressions in the 9000+ existing tests

## Follow-ups (tracked)

- **Phase 0 completion** — annotate `PPDS.Dataverse`, `PPDS.Migration`, `PPDS.Plugins` using the pilot pattern (larger scope than PPDS.Auth; not pre-documented)
- **GitHub App provisioning** — one-time setup per the runbook; workflow holds TODOs until secrets are configured
- **Test infrastructure leak** — `DocsReleaseWorkflowTests.DryRunProducesExpectedArtifacts` creates fake git repos in a temp dir but rogue "baseline" commits occasionally land in the real repo during pre-commit test runs. Tests pass correctly in isolation; the leak is benign (easy to reset) but should be fixed
- **BrowserHelper test isolation** — introduce `IBrowserOpener` interface so tests don't accidentally open real browser tabs via `Process.Start(url)` paths

## Verification status

- [x] /gates — mechanical build + test passes (see commit-per-phase full test runs in prior PR comments)
- [x] /verify workflow — all 8 structural checks pass (hook scripts, settings, skill/agent frontmatter, retro schema, behavioral scenarios)
- [x] /review — two bias-isolated review passes on spec + plan (51 findings triaged across them; all critical + important fixed)
- [ ] /qa — N/A for this PR; no product surface behavior changes, only tooling/generators
- [x] /pr monitor bugs fixed — PR will convert draft→ready correctly now

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by joshsmithxrm** [comment]
@gemini-code-assist review

File: .claude/skills/start/SKILL.md [hunk 1]
```diff
@@ -143,118 +143,147 @@ python scripts/workflow-state.py set launch_command "<confirmed-claude-command>"
...
 
-Use the launch command confirmed in Step 3.
+Routing instruction by launch command:
+- `claude` (bug fix) → "Reproduce the bug, write a regression test that fails, fix the code to make it pass, then run `/gates` → `/verify` → `/pr`."
+- `claude '/implement'` → "Read the spec and plan referenced above, then invoke `/implement` to execute end-to-end."
+- `claude '/design'` → "Invoke `/design` to size the work and route to `/spec` → `/plan` → `/implement`."
+- `claude` (docs) → "Edit the relevant docs and commit. No design or implement needed. When done: `/pr`."
+
+**Prompt rules:**
+- No single quotes in prompt content (here-string terminator sensitivity). Rephrase or use double quotes.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/start/SKILL.md]: Addressed in commit 119a20f15.

You're right — the blanket single-quote rule would corrupt every apostrophe ("it's", "don't", "Claude's") when capturing verbatim issue descriptions. Narrowed the rule to target only `'@` at the start of a line (the exact sequence that terminates PowerShell's single-quoted here-string `@' ... '@`). Mid-line apostrophes are now explicitly allowed in the guidance.

File: specs/docs-generation.md [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+
+### MdxEscape
+
+Shared utility used by all generators to produce MDX-safe markdown.
+
+```csharp
+public static class MdxEscape
+{
+    public static string Prose(string raw);    // < → &lt;, > → &gt; outside code fences
+    public static string InlineCode(string raw); // wraps in backticks, escapes backticks
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply specs/docs-generation.md]: Addressed in commit 119a20f15.

Correct finding — the previous 3-case logic had a CommonMark parsing bug: input `"foo \`bar\`"` was wrapped as `` \`\`foo \`bar\`\`\` ``, and the trailing 3-backtick run starts parsing as a fence marker, not a close delimiter.

`MdxEscape.InlineCode` now uses the CommonMark-canonical adaptive rule:
- Find the longest run of consecutive backticks in the content (N).
- Delimiter = (N+1) backticks — guaranteed not to appear inside.
- If the content starts or ends with a backtick, pad that side with a single space so the last/first backtick of content stays distinct from the delimiter run.

Dropped the fenced-block fallback — inline code is never multi-line in generator output, and longer adaptive delimiters cover every case (including `` \`\`\` `` in content → 4-backtick delimiter).

Tests: the prior `InlineCode_ContainsSingleBacktick_WrapsInDoubleBacktick` was asserting buggy output — rewritten to assert the correct padded form. Added coverage for double-backtick run, triple-backtick run (no fence fallback), and content that starts AND ends with a backtick. Spec one-liner at `specs/docs-generation.md` updated to describe the adaptive+padding contract.

File: scripts/docs-gen/libs-reflect/LibraryReferenceGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,598 @@
...
+                    Environment.GetFolderPath(Environment.SpecialFolder.UserProfile),
+                    ".nuget", "packages");
+            if (Directory.Exists(nugetHome))
+            {
+                try
+                {
+                    var matches = Directory.EnumerateFiles(
+                        nugetHome, name + ".dll", SearchOption.AllDirectories)
+                        .OrderBy(p => p, StringComparer.OrdinalIgnoreCase)
+                        .ToList();
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply scripts/docs-gen/libs-reflect/LibraryReferenceGenerator.cs]: Addressed in commit 7255c93da. Targeted probe of `{nugetHome}/{assemblyName}/` first (fast — bounded to one package subtree), full recursive scan only when the assembly simple name does not match its package id. Within matches, score (TFM-rank, version-rank) and prefer newest on both axes — fixes both concerns (full-cache traversal + older-version selection).

File: scripts/docs-gen/libs-reflect/LibraryReferenceGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,598 @@
...
+                    string? best = matches.FirstOrDefault(p => p.Contains("net8.0", StringComparison.Ordinal))
+                        ?? matches.FirstOrDefault(p => p.Contains("net9.0", StringComparison.Ordinal))
+                        ?? matches.FirstOrDefault(p => p.Contains("netstandard2.0", StringComparison.Ordinal))
+                        ?? matches.FirstOrDefault();
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply scripts/docs-gen/libs-reflect/LibraryReferenceGenerator.cs]: Addressed in commit 7255c93da. Replaced the null-coalescing chain with a `TfmRank` helper that scores net10.0 → net9.0 → net8.0 → netstandard2.0 descending (added net10.0 support). TFM match uses path-separator-bounded segments so "net10.0" does not spuriously match inside "net1.0".

File: scripts/docs-gen/app-token/Program.cs [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+    /// <summary>
+    /// Calls <c>GET /app/installations</c> and returns the installation id
+    /// whose <c>account.login</c> matches <paramref name="login"/>.
+    /// </summary>
+    /// <exception cref="InvalidOperationException">
+    /// Thrown when no matching installation is found.
+    /// </exception>
+    private static async Task<long> FindInstallationIdAsync(HttpClient http, string login)
+    {
+        using var response = await http.GetAsync("https://api.github.com/app/installations");
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply scripts/docs-gen/app-token/Program.cs]: Addressed in commit 7255c93da. `FindInstallationIdAsync` now iterates `page=1..N` at `per_page=100` and stops when a page returns fewer than `per_page` items. Hard cap at 100 pages (10,000 installations — defense against a malformed response looping forever).

File: scripts/docs-gen/libs-reflect/XmlDocIndex.cs [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+                            sb.Append('`').Append(langword).Append('`');
+                        }
+                        else if (!string.IsNullOrEmpty(cref))
+                        {
+                            var name = cref;
+                            var colon = name.IndexOf(':');
+                            if (colon >= 0) name = name.Substring(colon + 1);
+                            sb.Append('`').Append(name).Append('`');
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply scripts/docs-gen/libs-reflect/XmlDocIndex.cs:224]: Addressed in commit 7255c93da. After stripping the prefix, the cref name is run through `.Replace('{', '<').Replace('}', '>')` so readers see `List<T>` instead of `List{T}`. Libs-reflect tests 9/9 still green.

File: scripts/docs-gen/compute-surface-summary.sh [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+      ;;
+  esac
+done
+
+if [ -z "$ROOT" ]; then
+  script_dir="$(cd "$(dirname "$0")" && pwd)"
+  ROOT="$(cd "$script_dir/../.." && pwd)"
+fi
+
+cd "$ROOT"
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply scripts/docs-gen/compute-surface-summary.sh]: Addressed in commit 7255c93da. Dropped the `cd "$ROOT"`; both git calls now use `git -C "$ROOT" ...`. Absolute-path usage for `current="$ROOT/$shipped_rel"` was already correct, so removing the cd doesn't break file reads.
